GammaJul.ReSharper.ForTea/Psi/T4VBCodeGenerator.cs

```diff
@@ -0,0 +1,277 @@
+﻿#region License
+
+//    Copyright 2012 Julien Lebosquain
+//    Copyright 2016 Caelan Sayler - [caelantsayler]at[gmail]com
+// 
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+// 
+//        http://www.apache.org/licenses/LICENSE-2.0
+// 
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#endregion
+
+
+using System;
+using GammaJul.ReSharper.ForTea.Psi.Directives;
+using GammaJul.ReSharper.ForTea.Tree;
+using JetBrains.Annotations;
+using JetBrains.Application;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.Tree;
+
+namespace GammaJul.ReSharper.ForTea.Psi {
+
+	/// <summary>
+	/// This class generates a code-behind file from VB embedded statements and directives in the T4 file.
+	/// </summary>
+	internal sealed class T4VBCodeGenerator : IRecursiveElementProcessor {
```

Comment:
A lot of logic is duplicated from `T4CSharpCodeGenerator`. I think a base class can be created to abstract away language-specific generation code. Are you up to it? If not, I can do it after merging.